### PR TITLE
Implemented Task

### PR DESCRIPTION
**Pull request by KarinaKrinitskaya**

### CODE REVIEW

File: CSharpAdvanced/CSharpAdvanced/Program.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+#region Task_1
+
+List<string> flowers = new List<string>() { "Lycoris", "Rose", "Violet", "Daisy", "Cornflower", "Peony" };
+
+GetFirstElemList(flowers);
+
+static T GetFirstElemList<T>(List<T> someList)
+{
+    T firstElemInList = someList.First();
```
Comment (dnnyhlwdplc) [inline CSharpAdvanced/CSharpAdvanced/Program.cs]: return someList.First()

File: CSharpAdvanced/CSharpAdvanced/Program.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+static double GetHeapPriceLine<T>(T flower, int q) where T : Flower
+{
+    return flower.Price * q;
+}
+
+static void PrintHeapPriceLine<T>(T flower, int q) where T : Flower
+{
+    double totalPrice = GetHeapPriceLine<T>(flower, q);
+    Console.WriteLine($"{GetClassName<T>()} in quantity {q}, unit price {flower.Price}, total price {totalPrice}");
```
Comment (dnnyhlwdplc) [inline CSharpAdvanced/CSharpAdvanced/Program.cs]: why so hard? :) flower.Name :)

File: CSharpAdvanced/CSharpAdvanced/FlowerRepainter.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        };
+
+        public static void IncreasePrice(T flower)
+        {
+            flower.Price *= 1.15;
+        }
+
+        public static void RepaintFlower(T flower)
+        {
+            Console.WriteLine($"\nChoose the color for your {flower.Name}. Enter the number:");
```
Comment (dnnyhlwdplc) [inline CSharpAdvanced/CSharpAdvanced/FlowerRepainter.cs:23]: please add verification that if the color stays the same you will not increase price :)

File: CSharpAdvanced/CSharpAdvanced/BookExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        }
+
+        public static void OooooopsMethod(this Book book, int oopsCoef)
+        {
+            book.Pages = book.Pages * oopsCoef / 100;
+            Console.WriteLine($"Oooooops! Goblins has stolen pages from {book.Name}. {book.Name} has {book.Pages} pages now, We probably wont recover your money");
+        }
+
+        public static void ToHelloKitty(this Book book)
```
Comment (dnnyhlwdplc) [inline CSharpAdvanced/CSharpAdvanced/BookExtensions.cs:40]: I love that fact that you followed code standards and created extension class only for 1 data type!

**Review by dnnyhlwdplc — APPROVED** [review]

File: CSharpAdvanced/CSharpAdvanced/Program.cs (hunk 1 above)
Comment (KarinaKrinitskaya) [reply CSharpAdvanced/CSharpAdvanced/Program.cs]: Fixed

File: CSharpAdvanced/CSharpAdvanced/Program.cs (hunk 2 above)
Comment (KarinaKrinitskaya) [reply CSharpAdvanced/CSharpAdvanced/Program.cs]: fixed

File: CSharpAdvanced/CSharpAdvanced/FlowerRepainter.cs (hunk 3 above)
Comment (KarinaKrinitskaya) [reply CSharpAdvanced/CSharpAdvanced/FlowerRepainter.cs:23]: I have this verification on the 30 line:)

**Pull request merged**